p2p/libp2p/p2pSigner.go

```diff
@@ -0,0 +1,42 @@
+package libp2p
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	libp2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type p2pSigner struct {
+	privateKey *libp2pCrypto.Secp256k1PrivateKey
+}
+
+// Sign will sign a payload with the internal private key
+func (signer *p2pSigner) Sign(payload []byte) ([]byte, error) {
+	return signer.privateKey.Sign(payload)
```

Comment:
since this struct doesn't have a constructor, maybe check if `signer.privateKey` is nil?

Reply:
this is component was specifically made without a constructor because I wanted to be used as a sub-component. 
There is no reason for adding a nil check as that will be a setup error. The unit test should prevent this to happen.

Reply:
second, if we add a nil check there but we forgot to properly instantiate the netMessenger.p2pSigner the code will still panic as the signer.xxxxx will trigger a nil dereference exception. We would just clutter the code with useless nil checks that will never be triggered in a correctly setup component. The unit test(s)/future integration tests ensure that the code will not get damaged accidentally.